awscli/customizations/cloudfront.py

```diff
@@ -33,6 +33,36 @@ def register(event_handler):
         'operation-args-parsed.cloudfront.create-invalidation',
         validate_mutually_exclusive_handler(['invalidation_batch'], ['paths']))
 
+    event_handler.register(
+        'operation-args-parsed.cloudfront.create-distribution',
+        validate_mutually_exclusive_handler(
+            ['default_root_object', 'origin_domain_name'],
+            ['distribution_config']))
+    event_handler.register(
+        'building-argument-table.cloudfront.create-distribution',
+        lambda argument_table, **kwargs: argument_table.__setitem__(
```

Comment:
To document the off-line conversation, the `__setitem__` here is needed because it is inside a lambda body.